File: insert_ga.js
const fs = require('fs');
const path = require('path');

const gaCode = `<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-B1TE1JHHE3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-B1TE1JHHE3');
</script>
`;

function processDir(dir) {
    let count = 0;
    const items = fs.readdirSync(dir);
    for (const item of items) {
        if (item === '.gemini' || item === '.git' || item === 'node_modules') continue;
        const fullPath = path.join(dir, item);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            count += processDir(fullPath);
        } else if (item.endsWith('.html')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            if (!content.includes('G-B1TE1JHHE3')) {
                const newContent = content.replace('<head>', '<head>\n    ' + gaCode);
                if (newContent !== content) {
                    fs.writeFileSync(fullPath, newContent, 'utf8');
                    count++;
                    console.log('Updated:', fullPath);
                }
            } else {
                console.log('Already has GA:', fullPath);
            }
        }
    }
    return count;
}

const updated = processDir('.');
console.log('Total updated files:', updated);

File: insert_ga.py
import os
import glob

ga_code = """<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-B1TE1JHHE3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-B1TE1JHHE3');
</script>
"""

adsense_code = """<!-- Google AdSense -->
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-XXXXXXXXXXXXXXXX" crossorigin="anonymous"></script>
"""

count = 0
for filepath in glob.glob('**/*.html', recursive=True):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()

        new_content = content
        
        # Insert GA4
        if 'G-B1TE1JHHE3' not in new_content:
            new_content = new_content.replace('<head>', '<head>\n    ' + ga_code, 1)
            
        # Insert AdSense placeholder if not exist
        if 'ca-pub-XXXXXXXX' not in new_content and 'adsbygoogle.js' not in new_content:
             new_content = new_content.replace('</head>', '    ' + adsense_code + '</head>', 1)

        if new_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            count += 1
            print(f"Updated: {filepath}")
    except Exception as e:
        print(f"Error on {filepath}: {e}")

print(f"Total updated files: {count}")
